catalyst_voices/packages/internal/catalyst_voices_view_models/lib/src/campaign/campaign_info.dart

```diff
@@ -0,0 +1,110 @@
+import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:catalyst_voices_view_models/src/campaign/campaign_stage.dart';
+import 'package:equatable/equatable.dart';
+
+sealed class CampaignInfo extends Equatable {
+  final String id;
+  final CampaignStage stage;
+  final String description;
+
+  const CampaignInfo({
+    required this.id,
+    required this.stage,
+    required this.description,
+  });
+
+  /// Calculates the [campaign] info state at given [date].
+  factory CampaignInfo.fromCampaign(Campaign campaign, DateTime date) {
+    final stage = CampaignStage.fromCampaign(campaign, date);
+    return switch (stage) {
+      CampaignStage.draft => DraftCampaignInfo(
+          startDate: campaign.startDate,
+          id: campaign.id,
+          description: campaign.description,
+        ),
+      CampaignStage.scheduled => ScheduledCampaignInfo(
+          startDate: campaign.startDate,
+          id: campaign.id,
+          description: campaign.description,
+        ),
+      CampaignStage.live => LiveCampaignInfo(
+          startDate: campaign.startDate,
+          id: campaign.id,
+          description: campaign.description,
+        ),
+      CampaignStage.completed => CompletedCampaignInfo(
+          id: campaign.id,
+          description: campaign.description,
+        ),
+    };
+  }
+
+  @override
+  List<Object?> get props => [stage, description];
+}
+
+class ScheduledCampaignInfo extends CampaignInfo with CampaignDateTimeMixin {
+  @override
+  final DateTime startDate;
+
+  const ScheduledCampaignInfo({
+    required this.startDate,
+    required super.id,
+    required super.description,
+  }) : super(stage: CampaignStage.draft);
+
+  @override
+  List<Object?> get props => [startDate, description];
+}
+
+class DraftCampaignInfo extends CampaignInfo with CampaignDateTimeMixin {
+  @override
+  final DateTime startDate;
+
+  const DraftCampaignInfo({
+    required this.startDate,
+    required super.id,
+    required super.description,
+  }) : super(stage: CampaignStage.draft);
+
+  @override
+  List<Object?> get props => [startDate, description];
```

Comment:
with those props you're dismissing all props from CampaignInfo, are you sure about it?